### PR TITLE
Update javausage.md

### PR DESCRIPTION
**Pull request by kjanosz**
Change javausage.md to reflect the new Java API.

### CODE REVIEW

File: home/gettingstarted/0.7/javausage.md [hunk 1]
```diff
@@ -12,66 +12,117 @@ outof: 50
...
 
-Lets create a new file named `JavaRegressionTest2.java`.
+Lets create a new file named `MyHistogramTest.java`.
 We will use it to write an offline report test, which tests the performance of a histogram application.
 We will traverse a linked list and group its elements to different linked lists in a hash map.
 We start by importing the following package, which allows us to use the ScalaMeter's Java interface:
 
     import org.scalameter.japi.*;
+    
+Next, we import needed anntoations from the `annotation` subpackage:
```
Comment (axel22) [inline home/gettingstarted/0.7/javausage.md]: ...we import required annotations from the `annotation` subpackage:

File: home/gettingstarted/0.7/javausage.md [hunk 2]
```diff
@@ -12,66 +12,117 @@ outof: 50
...
-            .put("exec.outliers.suspectPercent", 40);
-
-Next, we declare a generator for the `groupBy` class (see more about generators in the [generator section](/home/gettingstarted/0.7/generators/index.html)):
-
-          public JavaGenerator<LinkedList<Integer>> generator() {
-            JavaGenerator<Integer> sizes = new SingleGen("size", 5000000);
-            return new CollectionGenerators(sizes).lists();
+In the Scala frontend, we encode the groups using nested closures.
+In the Java frontend, we encode them by placing the `benchmark` annotation 
+that argument denotes the group.
```
Comment (axel22) [inline home/gettingstarted/0.7/javausage.md]: ...whose `String` argument denotes the group that the benchmark belongs to.

File: home/gettingstarted/0.7/javausage.md [hunk 3]
```diff
@@ -12,66 +12,117 @@ outof: 50
...
+We now declare benchmark snippet stub in a `list` group:
+
+      @benchmark("list")
```
Comment (axel22) [inline home/gettingstarted/0.7/javausage.md]: To show that the groups can nest in this frontend too, I would suggest that you use a group name that contains dots (`.`).

File: home/gettingstarted/0.7/javausage.md [hunk 4]
```diff
@@ -12,66 +12,117 @@ outof: 50
...
+      public void histogram() {
+      
+      }
+      
+Next, we want a custom context:
```
Comment (axel22) [inline home/gettingstarted/0.7/javausage.md]: Next, we want to declare a custom configuration context:

File: home/gettingstarted/0.7/javausage.md [hunk 5]
```diff
@@ -12,66 +12,117 @@ outof: 50
...
+
+      public final Context config = new ContextBuilder()
+          .put("exec.benchRuns", 20)
+          .put("exec.independentSamples", 1)
+          .put("exec.outliers.covMultiplier", 1.5)
+          .put("exec.outliers.suspectPercent", 40)
+          .build();
+          
+We want it not only for our histogram method
```
Comment (axel22) [inline home/gettingstarted/0.7/javausage.md]: Let's assume that we want to apply this custom `config` not only to our `histogram` method,
    but to all benchmark snippets in the `list` group.
    We do this by adding the optional `scopes` annotation to the `MyHistogramTest` class, as follows:

File: home/gettingstarted/0.7/javausage.md [hunk 6]
```diff
@@ -12,66 +12,117 @@ outof: 50
...
+      ));
+      
+      @gen("lists")
+      @benchmark("list")
+      public void histogram(Tuple2<Integer, LinkedList<Integer>> v) {
+        
+      }
 
-Finally, we declare the code snippet for the test by implementing the `snippet` method:
+Now we initialize our list with random data in the setup method:
```
Comment (axel22) [inline home/gettingstarted/0.7/javausage.md]: Now, assume that the list needs to be initialized with random data before the benchmark begins.
    We will declare our own custom setup method named `listSetup` that does this initialization.
    Then, to ensure that `listSetup` is invoked before the benchmark,
    we need to add the `@setup` annotation to the `histogram` method:

File: home/gettingstarted/0.7/javausage.md [hunk 7]
```diff
@@ -12,66 +12,117 @@ outof: 50
...
+      }
+      
+      @gen("lists")
+      @setup("listSetup")
+      @benchmark("list")
+      public void histogram(Tuple2<Integer, LinkedList<Integer>> v) {
+        
+      }
+      
+Next, we would like to have custom curve name:
```
Comment (axel22) [inline home/gettingstarted/0.7/javausage.md]: By default, the curve corresponding to the benchmark snippet will be equal to the method name.
    If we would like a custom name for this method, we need to use the `@curve` annotation:

File: home/gettingstarted/0.7/javausage.md [hunk 8]
```diff
@@ -12,66 +12,117 @@ outof: 50
...
+
+      @gen("lists")
+      @setup("listSetup")
+      @curve("groupBy")
+      @benchmark("list")
+      public void histogram(Tuple2<Integer, LinkedList<Integer>> v) {
+        
+      }
+
+Finally, we implement the actual code snippet method:
```
Comment (axel22) [inline home/gettingstarted/0.7/javausage.md]: Finally, we add the actual code of the snippet method:

File: home/gettingstarted/0.7/javausage.md [hunk 9]
```diff
@@ -82,40 +133,58 @@ Finally, we declare the code snippet for the test by implementing the `snippet`
...
+      public final Context config = new ContextBuilder()
+          .put("exec.benchRuns", 20)
+          .put("exec.independentSamples", 1)
+          .put("exec.outliers.covMultiplier", 1.5)
+          .put("exec.outliers.suspectPercent", 40)
+          .build();
+    
+      public final JGen<Integer> sizes = JGen.intValues("size", 500000);
+    
+      public final JGen<Tuple2<Integer, LinkedList<Integer>>> lists = sizes.zip(sizes.map(
```
Comment (axel22) [inline home/gettingstarted/0.7/javausage.md]: We might need to wrap this line, because it might escape the <code> element in the generated HTML.

**Comment by kjanosz** [comment]
@axel22: Updated.

**Pull request merged**